src/Microsoft.DotNet.Interactive.Http/Model/HttpNamedRequest.cs

```diff
@@ -233,11 +234,32 @@ internal HttpNamedRequest(HttpRequestNode httpRequestNode, HttpResponse response
     {
         if (currentIndex + 1 == path.Length)
         {
-            var result = responseJSON[path[currentIndex]];
-            return result?.ToString();
+            if(responseJSON.GetType() == typeof(JsonArray))
+            {
+                var jsonArray = (JsonArray)responseJSON;
+                return jsonArray.FirstOrDefault(n => n?[path[currentIndex]] != null)?.ToJsonString();
+            } 
+            else if(responseJSON.GetType() == typeof(JsonObject))
```

Comment:
You can do similar here. You can also turn the whole block into a `switch`, e.g.

```csharp
switch (responseJson)
{
  case JsonArray jsonArray: //...
  case JsonObject jsonObject: // ...
}
```